### PR TITLE
analyzer: Upgrade Gradle test projects to Gradle 7.5.1

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Upgrade the Gradle test projects to Gradle 7.5.1 and update the configuration where required.

### CODE REVIEW

**Comment by sschuberth** [comment]
Don't we want to deliberately keep them at old versions to ensure we can still analyze such projects?

**Comment by mnonnenmacher** [comment]
> Don't we want to deliberately keep them at old versions to ensure we can still analyze such projects?

IMO no, we have a dedicated test for compatibility even though it is currently deactivated. And if we do compatibility tests it should happen with a strategy, not by testing arbitrary versions as part of tests which were created to test other functionality.

File: analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ analyzer:
     variables: {}
     tool_versions: {}
   config:
-    allow_dynamic_versions: false
+    allow_dynamic_versions: true
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml:50]: Any idea why this is `true` now? The default is `false`, and the only `OrtMainFunTest` which uses this file does not seem to modify the default...

Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml:50]: Maybe come local configuration of your is being taken into account, as the test runs the full `OrtMain`?

**Comment by mnonnenmacher** [comment]
@sschuberth FYI, I have started to work on activating the Gradle compatibility tests again by updating and reducing the size of the test.

File: analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml]: Indeed, fixed. This should not be the case, I think I'll make a separate PR to decouple the test from the user configuration.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**